mainProject/src/main/java/com/epam/ivanou4/vehicle/controller/CompanyRestController.java

```
@@ -32,21 +37,38 @@ public Company get(@PathVariable("id") String id) {
     }
 
     @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Company> createWithLocation(@RequestBody Company company) {
-        Company createdCompany = service.create(company);
-        URI uriOfNewResource = ServletUriComponentsBuilder.fromCurrentContextPath()
-                .path(REST_URL + "/{id}")
-                .buildAndExpand(createdCompany.getId()).toUri();
-        return ResponseEntity.created(uriOfNewResource).body(createdCompany);
+    public ResponseEntity<Company> createWithLocation(@RequestBody String companyJson) {
+        if (new Validation().validate(companyJson, COMPANY_SCHEMA)) {
+            Company company = mapCompany(companyJson);
+            Company createdCompany = service.create(company);
+            URI uriOfNewResource = ServletUriComponentsBuilder.fromCurrentContextPath()
+                    .path(REST_URL + "/{id}")
+                    .buildAndExpand(createdCompany.getId()).toUri();
+            return ResponseEntity.created(uriOfNewResource).body(createdCompany);
+        }
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
     }
 
     @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public void update(@RequestBody Company company) {
-        service.update(company);
+    public void update(@RequestBody String companyJson) {
+        if (new Validation().validate(companyJson, COMPANY_SCHEMA)) {
+            Company company = mapCompany(companyJson);
+            service.update(company);
+        }
     }
 
     @DeleteMapping(value = "/{id}")
     public void delete(@PathVariable("id") String id) {
         service.delete(id);
     }
+
+    private Company mapCompany(String json) {
+        ObjectMapper mapper = new ObjectMapper();
+        Company company = null;
+        try {
+            company = mapper.readValue(json, Company.class);
+        } catch (IOException ignored) {
```

Ignoring exception  is bad practise. Please log that in some way